### ISSUE TITLE
Optimize Gradient Descent

### ISSUE THREAD

**Issue by ZChosed** [cornellev/scan-matching#4]
Possible options to increase speed of convergence using gradient descent:
- fine tune the learning rate up (this is dangerous if gradients can grow very large)
- use something like adagrad to guarantee convergence on all data sets (even with exterme gradients)
- use newtons method when descent slows past some metric (defining this will be data dependent and may be hard to find) to converge to a local min instantly

**Comment by ethanuppal** [comment]
@zephanrs also suggested trying SGD

**Comment by sidharthmrao** [comment]
> @zephanrs also suggested trying SGD

I did too :pleading_face: :disappointed:

**Comment by sidharthmrao** [comment]
Maybe try sampling like 7 sets of 4 points in sequence or something when computing loss

**Comment by ethanuppal** [comment]
> > @zephanrs also suggested trying SGD
> 
> 
> 
> I did too :pleading_face: :disappointed: 

Well you don't matter

**Comment by sidharthmrao** [comment]
> > > @zephanrs also suggested trying SGD
> > 
> > 
> > 
> > I did too :pleading_face: :disappointed: 
> 
> Well you don't matter 

![image](https://github.com/cornellev/scan-matching/assets/24660439/7319f614-3418-49b9-b8e9-ac3bf6350505)
